src/main/java/uk/gov/justice/hmpps/prison/service/ImageService.java

```diff
@@ -47,5 +51,41 @@ public ImageDetail findImageDetail(final Long imageId) {
             .map(i -> fullSizeImage ? i.getFullSizeImage() : i.getThumbnailImage());
     }
 
+    @PreAuthorize("hasRole('SYSTEM_USER')")
+    @HasWriteScope
+    @Transactional
+    public ImageDetail putImageForOffender(final String offenderNumber, final boolean fullSizeImage,  final String imageData) {
+        // Check that the offender number exists
+        List<Offender> offenderList = offenderRepository.findByNomsId(offenderNumber);
```

Comment:
Use `offenderRepository.findOffenderByNomsId`

Reply:
JPA version

Reply:
`offenderRepository.findOffenderByNomsId(offenderNo)            .orElseThrow(EntityNotFoundException.withMessage(format("No prisoner found for prisoner number %s", offenderNo)));`

Reply:
Done